web/themes/new_weather_theme/assets/js/charts/qpf.js

```diff
@@ -0,0 +1,148 @@
+import { drawChart } from "./WeatherChart.js";
+import styles from "../styles.js";
+
+const round = (number, decimals) =>
+  Math.round(number * 10 ** decimals) / 10 ** decimals;
+
+const chartContainers = Array.from(
+  document.querySelectorAll(".wx-qpf-chart-container"),
+);
+
+const makePattern = async (imageUrl, size = 60) =>
+  new Promise((resolve) => {
+    const imgage = new Image();
+    imgage.src = imageUrl;
+    imgage.onload = () => {
```

Comment:
I'm imgagining that was a typo
```suggestion
    const image = new Image();
    image.src = imageUrl;
    image.onload = () => {
```

Reply:
Don't let your imgagination run away with you.